File: TrackMyAct.Server/Models/Entities/EventEntity.cs
namespace TrackMyAct.Server.Models.Entities;

public class EventEntity : Entity
{
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime StartDateTime { get; set; }
    public bool NeedAcception { get; set; }

    public PlaceEntity EventPlace { get; set; }
    public int EventPlaceId { get; set; }

    public OrganizerEntity Organizer { get; set; }
    public int OrganizerId { get; set; }
    public ICollection<ParticipantEntity> AcceptedParticipants { get; set; }

    public ICollection<ParticipantEntity> ParticipantRequests { get; set; }
    public ICollection<ParticipantEntity> OrganizerRequests { get; set; }
}

File: TrackMyAct.Mobile/composeApp/src/commonMain/kotlin/org/js/tma/dto/EventEntity.kt
package org.js.tma.dto

import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable

@Serializable
data class EventEntity(
    override var id: Int,
    var name: String,
    var description: String,
    var startDateTime: LocalDateTime,
    var needAcception: Boolean,

    var eventPlace: PlaceEntity?,
    var eventPlaceId: Int,

    var organizer: OrganizerEntity?,
    var organizerId: Int,

    var acceptedParticipants: List<ParticipantEntity>? = emptyList(),
    var participantRequests: List<ParticipantEntity>? = emptyList(),
    var organizerRequests: List<ParticipantEntity>? = emptyList()
) : Entity()